test/ctia/test_helpers/crud.clj

```diff
@@ -117,25 +122,101 @@
+                          (map #(str "updated-nil-false" %)
+                               (range 100)))
+                (is (= "updated-nil-true" (get-modified put nil "updated-nil-true")))))))
+
+          (when patch-tests?
+            (is (= "patched-true"
+                   (get-modified patch true "patched-true"))
+                "Patch queries should wait for index refresh when wait_for is true")
+            ;; we trigger next 404 tests several times because the refresh could occur between the first POST / GET sequence.
+            (is (some #(not= % (get-modified patch false %))
+                      (map #(str "patched-false" %)
+                           (range 100)))
+                "Patch queries should not wait for index refresh when wait_for is false")
+            (testing "Configured ctia.store.es.default.refresh value is applied when wait_for is not specified"
+              (if (#{"false" nil} (get-in @properties [:ctia :store :es :default :refresh]))
+                (is (some #(not= % (get-modified patch nil %))
+                          (map #(str "patched-nil-false" %)
+                               (range 100)))
+                    (is (= "patched-nil-true" (get-modified patch nil "updated-nil-true")))))))))
+
+      (testing "testing wait_for values on entity deletion"
+        (let [get-status (fn [wait_for]
+                           (let [entity-id (-> (post (str "ctia/" entity "?wait_for=true")
+                                                     :body new-record
+                                                     :headers headers)
+                                               :parsed-body
+                                               entity->short-id)
+                                 path (cond-> (format "ctia/%s/%s" entity entity-id)
+                                        (boolean? wait_for) (str "?wait_for=" wait_for))]
+                             (delete path
+                                     :headers headers)
+                             (-> (get (format "ctia/%s/%s" entity entity-id)
+                                      :headers headers)
+                                 :status)))]
+          (is (= 404 (get-status true))
+              "Delete queries should wait for index refresh when wait_for is true")
+          ;; we trigger next 200 tests several times because the refresh could occur between the first POST / GET sequence.
+          (is (some #(= 200 %)
+                    (repeatedly 3 #(get-status false)))
+             "Delete queries should not wait for index refresh when wait_for is false")
+          (testing "Configured ctia.store.es.default.refresh value is applied when wait_for is not specified"
+            (if (= "false" (get-in @properties [:ctia :store :es :default :refresh]))
+
+              (is (some #(= 200 %)
+                        (repeatedly 3 #(get-status nil))))
+              (is (= 404 (get-status nil)))))))
+
```

Comment:
I would only use `cli-http-fake` to check if the refresh param is correctly set in the ElasticSearch request